activerecord/test/models/admin/region.rb

```diff
@@ -0,0 +1,6 @@
+# The class Region (without namespace) shouldn't exist
+# in order to the test test_type_mismatch_with_namespaced_class
+# be valid
+class Admin::Region < ActiveRecord::Base
```

Comment:
Yes, defining `Admin::Region` in the test works, I'm updating the code, thanks :)